modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -2031,8 +2031,9 @@ void ONNXImporter::parseGemm(LayerParams& layerParams, const opencv_onnx::NodePr
     addLayer(layerParams, node_proto);
 }
 
-void ONNXImporter::parseMatMul(LayerParams& layerParams, const opencv_onnx::NodeProto& node_proto)
+void ONNXImporter::parseMatMul(LayerParams& layerParams, const opencv_onnx::NodeProto& node_proto_)
 {
+    opencv_onnx::NodeProto node_proto = node_proto_;
```

Comment:
const reference in signature and copy call on the first line looks very strange. I propose to pass `node_proto` as value, or, better, make a copy in else branch only to not degrade existing branch performance.

Reply:
Agree, this `const reference` exists in many onnx op codes.